File: app/src/main/java/com/lzm/lightLive/util/DialogUtil.java
package com.lzm.lightLive.util;

import android.content.Intent;
import android.graphics.Color;
import android.net.Uri;
import android.text.SpannableString;
import android.text.Spanned;
import android.text.TextUtils;
import android.text.style.ForegroundColorSpan;
import android.text.style.UnderlineSpan;
import android.util.Log;
import android.view.View;
import android.widget.TextView;

import androidx.appcompat.widget.AppCompatTextView;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.airbnb.lottie.LottieAnimationView;
import com.bumptech.glide.Glide;
import com.kongzue.dialogx.dialogs.FullScreenDialog;
import com.kongzue.dialogx.interfaces.OnBindView;
import com.lzm.lightLive.R;
import com.lzm.lightLive.adapter.HostAdapter;
import com.lzm.lightLive.common.Const;
import com.lzm.lightLive.http.BaseResult;
import com.lzm.lightLive.http.RetrofitManager;
import com.lzm.lightLive.http.bean.Room;
import com.lzm.lightLive.http.bean.dy.DyCate;
import com.lzm.lightLive.http.bean.dy.DySortRoom;
import com.lzm.lightLive.http.request.dy.DyHttpRequest;
import com.lzm.lightLive.view.RoundImageView;

import java.util.ArrayList;
import java.util.List;

import io.reactivex.Observer;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.Disposable;
import io.reactivex.schedulers.Schedulers;

public class DialogUtil {

    private static final String TAG = "DialogUtil";

    public static void showHostInfoDialog(Room roomInfo) {

        FullScreenDialog.show(new OnBindView<FullScreenDialog>(R.layout.layout_dialog_host) {
            @Override
            public void onBind(FullScreenDialog dialog, View v) {
                RoundImageView avatar = v.findViewById(R.id.iv_avatar);
                AppCompatTextView hostName = v.findViewById(R.id.tv_host_name);
                AppCompatTextView roomName = v.findViewById(R.id.tv_room_name);
                AppCompatTextView roomId = v.findViewById(R.id.tv_room_id);
                AppCompatTextView cateName = v.findViewById(R.id.tv_cate_name);
                AppCompatTextView roomUrl = v.findViewById(R.id.tv_room_url);
                AppCompatTextView liveUri = v.findViewById(R.id.tv_live_uri);
                LottieAnimationView animView = v.findViewById(R.id.animView);

                Glide.with(v)
                        .load(roomInfo.getHostAvatar())
                        .into(avatar);
                hostName.setText(roomInfo.getHostName());
                roomName.setText(roomInfo.getRoomName());
                roomId.setText(roomInfo.getRoomId());
                cateName.setText(roomInfo.getCateName());
                String url = "";
                switch (roomInfo.getPlatform()) {
                    case Room.LIVE_PLAT_DY:
                        url = "https://www.douyu.com/" + roomInfo.getRoomId();
                        break;
                    case Room.LIVE_PLAT_HY:
                        url = "https://www.huya.com/" + roomInfo.getRoomId();
                        break;
                    case Room.LIVE_PLAT_BL:
                        url = "https://live.bilibili.com/" + roomInfo.getRoomId();
                        break;
                }
                SpannableString string = new SpannableString(url);
                string.setSpan(new UnderlineSpan(), 0, string.length(), Spanned.SPAN_INCLUSIVE_EXCLUSIVE);
                string.setSpan(new ForegroundColorSpan(Color.parseColor("#1144aa")),
                        0, string.length(), Spanned.SPAN_INCLUSIVE_EXCLUSIVE);
                roomUrl.setText(string);

                roomUrl.setOnClickListener(tv->{
                    if(!TextUtils.isEmpty(roomUrl.getText())) {
                        Uri uri = Uri.parse(roomUrl.getText().toString());
                        Intent intent = new Intent(Intent.ACTION_VIEW, uri);
                        roomUrl.getContext().startActivity(intent);
                    }
                });

                liveUri.setText(roomInfo.getLiveStreamUriHigh());

                animView.setAnimation(R.raw.blue_bg);
                animView.setSpeed(2f);
                animView.playAnimation();
            }
        });
    }

    public static void showCateInfoDialog(DyCate.DyCateData cateData) {

        FullScreenDialog.show(new OnBindView<FullScreenDialog>(R.layout.layout_dialog_cate) {
            @Override
            public void onBind(FullScreenDialog dialog, View v) {
                TextView tvCateName = v.findViewById(R.id.tv_cate_name);
                RecyclerView recyclerView = v.findViewById(R.id.ry_cate);
                recyclerView.setLayoutManager(new GridLayoutManager(v.getContext(), 2));
                tvCateName.setText(cateData.getCateName());

                List<Room> roomList = new ArrayList<>();
                HostAdapter mAdapter = new HostAdapter(R.layout.layout_item_host, roomList);
                mAdapter.setPageMaxItem(50);
                recyclerView.setAdapter(mAdapter);

                DyHttpRequest call = RetrofitManager.getDyRetrofit().create(DyHttpRequest.class);

                call.queryCateHosts(cateData.getCateId(), 1)
                        .subscribeOn(Schedulers.io())
                        .observeOn(AndroidSchedulers.mainThread())
                        .subscribe(new Observer<BaseResult<DySortRoom>>() {
                            @Override
                            public void onSubscribe(Disposable d) { }

                            @Override
                            public void onNext(BaseResult<DySortRoom> result) {
                                Log.e(TAG, "onNext: " + result.getData() );
                                if(null != result.getData()) {
                                    List<Room> roomList = new ArrayList<>();
                                    for (DySortRoom.Related room : result.getData().getRelatedList()) {
                                        Room _temp = new Room(String.valueOf(room.getRoomId()), Room.LIVE_PLAT_DY);
                                        _temp.setRoomId(String.valueOf(room.getRoomId()));
                                        _temp.setCateName(room.getCateName());
                                        _temp.setHeatNum(room.getOnline());
                                        _temp.setStreamStatus(Room.LIVE_STATUS_ON);
                                        _temp.setRoomName(room.getRoomName());
                                        _temp.setHostName(room.getNickName());
                                        String prefix = ".jpg";
                                        if(room.getAvatar().contains("avatar")
                                                || room.getAvatar().contains("avanew")) {
                                            prefix = "_big.jpg";

                                        }
                                        _temp.setHostAvatar(Const.AVATAR_URL_DY + room.getAvatar() + prefix);
                                        _temp.setThumbUrl(room.getThumb());
                                        roomList.add(_temp);
                                    }
                                    mAdapter.setMassiveData(roomList);
                                }
                            }

                            @Override
                            public void onError(Throwable e) {
                                Log.e(TAG, "onError: " + e.getMessage() );
                            }

                            @Override
                            public void onComplete() { }
                        });
            }
        });
    }

}

File: app/src/main/java/com/lzm/lightLive/util/DialogUtil.kt
package com.lzm.lightLive.util

import android.content.Intent
import android.graphics.*
import android.net.Uri
import android.text.SpannableString
import android.text.Spanned
import android.text.TextUtils
import android.text.style.ForegroundColorSpan
import android.text.style.UnderlineSpan
import android.util.Log
import android.view.*
import android.widget.TextView
import androidx.appcompat.widget.AppCompatTextView
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.airbnb.lottie.LottieAnimationView
import com.bumptech.glide.Glide
import com.kongzue.dialogx.dialogs.FullScreenDialog
import com.kongzue.dialogx.interfaces.OnBindView
import com.lzm.lightLive.R
import com.lzm.lightLive.adapter.HostAdapter
import com.lzm.lightLive.common.Const
import com.lzm.lightLive.http.BaseResult
import com.lzm.lightLive.http.RetrofitManager
import com.lzm.lightLive.http.bean.Room
import com.lzm.lightLive.http.bean.dy.DyCate.DyCateData
import com.lzm.lightLive.http.bean.dy.DySortRoom
import com.lzm.lightLive.http.request.dy.DyHttpRequest
import com.lzm.lightLive.view.RoundImageView
import io.reactivex.Observer
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.Disposable
import io.reactivex.schedulers.Schedulers

object DialogUtil {
    private const val TAG = "DialogUtil"
    fun showHostInfoDialog(roomInfo: Room?) {
        FullScreenDialog.show(object : OnBindView<FullScreenDialog?>(R.layout.layout_dialog_host) {
            override fun onBind(dialog: FullScreenDialog?, v: View) {
                val avatar = v.findViewById<RoundImageView>(R.id.iv_avatar)
                val hostName = v.findViewById<AppCompatTextView>(R.id.tv_host_name)
                val roomName = v.findViewById<AppCompatTextView>(R.id.tv_room_name)
                val roomId = v.findViewById<AppCompatTextView>(R.id.tv_room_id)
                val cateName = v.findViewById<AppCompatTextView>(R.id.tv_cate_name)
                val roomUrl = v.findViewById<AppCompatTextView>(R.id.tv_room_url)
                val liveUri = v.findViewById<AppCompatTextView>(R.id.tv_live_uri)
                val animView = v.findViewById<LottieAnimationView>(R.id.animView)
                Glide.with(v)
                    .load(roomInfo?.hostAvatar)
                    .into(avatar)
                hostName.text = roomInfo?.hostName
                roomName.text = roomInfo?.roomName
                roomId.text = roomInfo?.roomId
                cateName.text = roomInfo?.cateName
                var url = ""
                when (roomInfo?.platform) {
                    Room.LIVE_PLAT_DY -> url =
                        "https://www.douyu.com/" + roomInfo.roomId
                    Room.LIVE_PLAT_HY -> url =
                        "https://www.huya.com/" + roomInfo.roomId
                    Room.LIVE_PLAT_BL -> url =
                        "https://live.bilibili.com/" + roomInfo.roomId
                }
                val string = SpannableString(url)
                string.setSpan(UnderlineSpan(), 0, string.length, Spanned.SPAN_INCLUSIVE_EXCLUSIVE)
                string.setSpan(
                    ForegroundColorSpan(Color.parseColor("#1144aa")),
                    0, string.length, Spanned.SPAN_INCLUSIVE_EXCLUSIVE
                )
                roomUrl.text = string
                roomUrl.setOnClickListener {
                    if (!TextUtils.isEmpty(roomUrl.text)) {
                        val uri = Uri.parse(roomUrl.text.toString())
                        val intent = Intent(Intent.ACTION_VIEW, uri)
                        roomUrl.context.startActivity(intent)
                    }
                }
                liveUri.text = roomInfo?.liveStreamUriHigh
                animView.setAnimation(R.raw.blue_bg)
                animView.speed = 2f
                animView.playAnimation()
            }
        })
    }

    fun showCateInfoDialog(cateData: DyCateData) {
        FullScreenDialog.show(object : OnBindView<FullScreenDialog?>(R.layout.layout_dialog_cate) {
            override fun onBind(dialog: FullScreenDialog?, v: View) {
                val tvCateName = v.findViewById<TextView>(R.id.tv_cate_name)
                val recyclerView = v.findViewById<RecyclerView>(R.id.ry_cate)
                recyclerView.layoutManager = GridLayoutManager(v.context, 2)
                tvCateName.text = cateData.cateName
                val roomList: List<Room> = ArrayList()
                val mAdapter = HostAdapter(R.layout.layout_item_host, roomList)
                mAdapter.pageMaxItem = 50
                recyclerView.adapter = mAdapter
                val call = RetrofitManager.dyRetrofit.create(
                    DyHttpRequest::class.java
                )
                call.queryCateHosts(cateData.cateId, 1)
                    .subscribeOn(Schedulers.io())
                    .observeOn(AndroidSchedulers.mainThread())
                    .subscribe(object : Observer<BaseResult<DySortRoom?>> {
                        override fun onSubscribe(d: Disposable) {}
                        override fun onNext(result: BaseResult<DySortRoom?>) {
                            if (null != result.data) {
                                val roomList = ArrayList<Room?>()
                                for (room in result.data?.relatedList!!) {
                                    val temp =
                                        Room(room.roomId.toString(), Room.LIVE_PLAT_DY)
                                    temp.roomId = room.roomId.toString()
                                    temp.cateName = room.cateName
                                    temp.heatNum = room.online
                                    temp.streamStatus = Room.LIVE_STATUS_ON
                                    temp.roomName = room.roomName
                                    temp.hostName = room.nickName
                                    var prefix = ".jpg"
                                    if (!TextUtils.isEmpty(room.avatar)
                                        && room.avatar!!.contains("avatar")
                                        || room.avatar!!.contains("avanew")
                                    ) {
                                        prefix = "_big.jpg"
                                        temp.hostAvatar =
                                            Const.AVATAR_URL_DY + room.avatar + prefix
                                    }
                                    if (!TextUtils.isEmpty(room.thumb)) {
                                        temp.thumbUrl = room.thumb
                                    }
                                    roomList.add(temp)
                                }
                                mAdapter.setMassiveData(roomList)
                            }
                        }

                        override fun onError(e: Throwable) {
                            Log.e(TAG, "onError: " + e.message)
                        }

                        override fun onComplete() {}
                    })
            }
        })
    }
}